src/lib/file/stream.mjs

```diff
@@ -0,0 +1,69 @@
+import { Readable } from 'stream';
+import StormLib from '../stormlib';
+
+const STREAM_CHUNK_SIZE = 64 * 1024;
+
+class FileStream extends Readable {
+  constructor(file) {
+    super({
+      highWaterMark: STREAM_CHUNK_SIZE
+    });
+
+    this._file = file;
+    this._size = file.size;
+    this._pos = 0;
+    this._buffer = new StormLib.Buf(Math.min(STREAM_CHUNK_SIZE, this._size));
+
+    this.on('end', () => {
+      this.destroy();
+    });
```

Comment:
Does this event handler need to be cleaned up on destroy?

Reply:
I don't believe so. The event handler is attached to `FileStream` (ie itself), so when JS GC kicks in, it'll be swept up.

Related note: we have to use `destroy` to clean up the C++-side stuff, since there's no finalizer hook in JS.

Reply:
Ah right! I seemed to remember certain closures could keep an object referenced, but likely some old browser quirk. Let's keep it as it is for now 👍